x/upgrade/keeper/keeper.go

```diff
@@ -76,89 +74,87 @@ func (k Keeper) ScheduleUpgrade(ctx sdk.Context, plan types.Plan) error {
 	}
 
 	store := ctx.KVStore(k.storeKey)
+
+	// clear any old IBC state stored by previous plan
+	oldPlan, exists := k.GetUpgradePlan(ctx)
+	if exists && oldPlan.IsIBCPlan() {
+		k.ClearIBCState(ctx, oldPlan.Height-1)
+	}
+
 	bz := k.cdc.MustMarshalBinaryBare(&plan)
 	store.Set(types.PlanKey(), bz)
 
-	if plan.UpgradedClientState == nil {
-		// if latest UpgradedClientState is nil, but upgraded client exists in store,
-		// then delete client state from store.
-		_, height, _ := k.GetUpgradedClient(ctx)
-		if height != 0 {
-			store.Delete(types.UpgradedClientKey(height))
+	if plan.IsIBCPlan() {
+		// Set UpgradedClientState in store
+		clientState, err := clienttypes.UnpackClientState(plan.UpgradedClientState)
+		if err != nil {
+			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "could not unpack clientstate: %v", err)
 		}
-		return nil
+		// deletes any previously stored upgraded client and sets the new upgraded client in
```

Comment:
looks like it no longer deletes any previously stored upgraded client since this happens the step before